main_test.go

```diff
@@ -402,6 +402,90 @@ func TestGenerate(t *testing.T) {
       "zone": "uscentral-5"
     }
   }
+}`,
+		},
+
+		{
+			desc: "happy case with v3 defaults and federation support enabled",
+			input: configInput{
+				xdsServerUri:             "example.com:443",
+				gcpProjectNumber:         123456789012345,
+				vpcNetworkName:           "thedefault",
+				ip:                       "10.9.8.7",
+				zone:                     "uscentral-5",
+				includeV3Features:        true,
+				includeFederationSupport: true,
+			},
+			wantOutput: `{
+  "xds_servers": [
+    {
+      "server_uri": "example.com:443",
+      "channel_creds": [
+        {
+          "type": "google_default"
+        }
+      ],
+      "server_features": [
+        "xds_v3"
+      ]
+    }
+  ],
+  "authorities": {
+    "": {},
+    "trafficdirector.googleapis.com:443": {}
+  },
+  "node": {
+    "id": "projects/123456789012345/networks/thedefault/nodes/9566c74d-1003-4c4d-bbbb-0407d1e2c649",
+    "cluster": "cluster",
+    "metadata": {
+      "INSTANCE_IP": "10.9.8.7",
+      "TRAFFICDIRECTOR_GCP_PROJECT_NUMBER": "123456789012345",
+      "TRAFFICDIRECTOR_NETWORK_NAME": "thedefault"
+    },
+    "locality": {
+      "zone": "uscentral-5"
+    }
+  }
+}`,
+		},
+
+		{
+			desc: "happy case with v2 config and federation support enabled",
+			input: configInput{
+				xdsServerUri:             "example.com:443",
+				gcpProjectNumber:         123456789012345,
+				vpcNetworkName:           "thedefault",
+				ip:                       "10.9.8.7",
+				zone:                     "uscentral-5",
+				includeV3Features:        false,
+				includeFederationSupport: true,
+			},
+			wantOutput: `{
+  "xds_servers": [
+    {
+      "server_uri": "example.com:443",
+      "channel_creds": [
+        {
+          "type": "google_default"
+        }
+      ]
+    }
+  ],
+  "authorities": {
+    "": {},
+    "trafficdirector.googleapis.com:443": {}
+  },
+  "node": {
+    "id": "52fdfc07-2182-454f-963f-5f0f9a621d72~10.9.8.7",
+    "cluster": "cluster",
+    "metadata": {
+      "TRAFFICDIRECTOR_GCP_PROJECT_NUMBER": "123456789012345",
+      "TRAFFICDIRECTOR_NETWORK_NAME": "thedefault"
+    },
+    "locality": {
+      "zone": "uscentral-5"
+    }
+  }
```

Comment:
I don't think we need this case as none of the logic added as part of this PR depends on whether or not v3 features are enabled.

Reply:
yup that makes sense to me